### PR TITLE
Add rails 7.1 support

### PR DESCRIPTION
**Pull request by reeganviljoen**
### What are you trying to accomplish?
Rails 7.1 dropped recently, it would be a good idea for view_component to keep up with these developments
https://guides.rubyonrails.org/7_1_release_notes.html

### What approach did you choose and why?

Add rails 7.1 to CI, appraisal and gemfile

### CODE REVIEW

**Comment by reeganviljoen** [comment]
@Spone can you please review

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,6 +38,8 @@ jobs:
             mode: "capture_patch_enabled"
           - rails_version: "7.0.2.3"
             ruby_version: "3.0"
```
Comment (reeganviljoen) [inline .github/workflows/ci.yml:40]:
```suggestion
          - rails_version: "7.0.2.3"
            ruby_version: "3.0"
            mode: "capture_patch_enabled"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -56,6 +59,8 @@ jobs:
             mode: "capture_patch_disabled"
           - rails_version: "7.0.2.3"
             ruby_version: "3.0"
```
Comment (reeganviljoen) [inline .github/workflows/ci.yml:61]:
```suggestion
          - rails_version: "7.0.2.3"
            ruby_version: "3.0"
            mode: "capture_patch_disabled"
```

File: Appraisals [hunk 3]
```diff
@@ -28,6 +28,11 @@ appraise "rails-7.0" do
   gem "tailwindcss-rails", "~> 2.0"
 end
 
+appraise "rails-7.1" do
+  gem "rails", "~> 7.1.1"
```
Comment (Spone) [inline Appraisals:32]: I think we can use the base version here (similarly to what we do for older versions above in the same file)

```suggestion
  gem "rails", "~> 7.1.0"
```

File: gemfiles/rails_7.1.gemfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# This file was generated by Appraisal
+
+source "https://rubygems.org"
+
+gem "capybara", "~> 3"
+gem "rails", "~> 7.1.1"
```
Comment (Spone) [inline gemfiles/rails_7.1.gemfile:6]: Please run `bundle exec appraisal generate` again, so this file is generated with your changes to `Appraisals`. Then, commit the changes.

**Comment by reeganviljoen** [comment]
@Spone I see the step that runs before is a gemfile cache step , this may be caching it to 7.0 would you be able to see if removing the cache fixes it

**Comment by Spone** [comment]
@reeganviljoen I cleared the cache and rerun the failing task, but it's still failing.

**Comment by Spone** [comment]
I think I figured it out: some ENV vars were still set to 7.0 in CI config files.

**Review by Spone — APPROVED** [review]

**Comment by Spone** [comment]
Now there's an issue with `Test compatibility with Primer ViewComponents` :/

**Comment by reeganviljoen** [comment]
@Spone it seems the version of webpacker being used is incompatible with `rails 7.1`, @camertron , @BlakeWilliams can you possibly investigate the issue on primer ?

**Comment by camertron** [comment]
Hey @reeganviljoen, apologies for not noticing your PR until now :/ I've [got one going](https://github.com/ViewComponent/view_component/pull/1889) as well. The PVC build issues are happening because we recently added 7.1 support over there as well, so we need to set `RAILS_VERSION: 7.1.0` for the PVC build job.

This issue is preventing me from releasing view_component, so I'm going to close this PR in the interest of expediting the release. I've credited you with the upgrade, as I noticed a few things in your PR that I had forgotten in mine 😄

**Pull request closed without merging**